api/v1alpha1/ext_auth_types.go

```diff
@@ -0,0 +1,97 @@
+)
+
+// ExtAuth defines the configuration for External Authorization.
+type ExtAuth struct {
+	// Type decides the type of External Authorization.
+	// Valid ExtAuthServiceType values are "GRPC" or "HTTP".
+	// +kubebuilder:validation:Enum=GRPC;HTTP
+	// +unionDiscriminator
+	Type ExtAuthServiceType `json:"type"`
+
+	// GRPC defines the gRPC External Authorization service
+	// Only one of GRPCService or HTTPService may be specified.
+	GRPC *GRPCExtAuthService `json:"grpcService,omitempty"`
+
+	// HTTP defines the HTTP External Authorization service
+	// Only one of GRPCService or HTTPService may be specified.
+	HTTP *HTTPExtAuthService `json:"httpService,omitempty"`
+
+	// AllowedHeaders defines the client request headers that will be included
+	// in the request to the external authorization service.
+	// Note: If not specified, the default behavior of different external authorization
+	// services is different. All headers will be included in the check request
+	// to a gRPC authorization server, whereas no headers will be included in the
+	// check request to an HTTP authorization server.
+	AllowedHeaders []string `json:"allowedHeaders,omitempty"`
+}
+
+// GRPCExtAuthService defines the gRPC External Authorization service
+type GRPCExtAuthService struct {
+	// Host is the hostname of the gRPC External Authorization service
+	Host gwapiv1a2.Hostname `json:"host"`
+
+	// Port is the network port of the gRPC External Authorization service
+	Port gwapiv1a2.PortNumber `json:"port"`
+
+	// TLS defines the TLS configuration for the gRPC External Authorization service.
+	// Note: If not specified, the proxy will talk to the gRPC External
+	// Authorization service in plaintext.
+	TLS *TLSConfig `json:"tlsSettings,omitempty"`
+}
+
+// HTTPExtAuthService defines the HTTP External Authorization service
+type HTTPExtAuthService struct {
+	// URL is the URL of the HTTP External Authorization service.
+	// The URL must be a fully qualified URL with a scheme, hostname,
+	// and optional port and path. Parameters are not allowed.
+	// The URL must use either the http or https scheme.
+	// If port is not specified, 80 for http and 443 for https are assumed.
+	// If path is specified, the authorization request will be sent to that path,
+	// or else the authorization request will be sent to the root path.
+	URL string `json:"url" yaml:"url"`
+
+	// TLS defines the TLS configuration for the HTTP External Authorization service.
+	// TLS is only valid when the URL scheme is https. If the URL scheme is
+	// https, and TLS is not specified, the proxy will use the system default
+	// certificate pool to verify the server certificate.
+	TLS *TLSConfig `json:"tlsSettings,omitempty"`
+
+	// Authorization response headers that will be added to the original client request
+	// before sending it to the upstream server.
+	// Note that coexistent headers will be overridden.
+	AllowedUpstreamHeaders []string `json:"allowedUpstreamHeaders,omitempty"`
```

Comment:
prefer using `Backend` instead of `Upstream` since that term is used everywhere in the upstream and EG API
also thoughts on these field names `HeadersToBackend` and `HeadersToExtAuth` / `HeadersToAuthService`

Reply:
How about renaming `AllowedUpstreamHeaders` To `AuthServiceHeadersToBackend`  and `AllowedHeaders` To `ClientHeadersToAuthService`?

Reply:
its very long ......
@envoyproxy/gateway-maintainers @envoyproxy/gateway-reviewers any suggestions ?

Reply:
Settled with  `AllowedBackendHeaders` until we figure out a better name.